pkg/sqlquery/scraper.go

```diff
@@ -29,25 +29,40 @@ import (
 	"go.uber.org/zap"
 )
 
-type scraper struct {
+type DbProviderFunc func() (*sql.DB, error)
+
+type ClientProviderFunc func(*sql.DB, string, *zap.Logger) DbClient
+
+type Scraper struct {
 	id                 config.ComponentID
 	query              Query
 	scrapeCfg          scraperhelper.ScraperControllerSettings
 	startTime          pcommon.Timestamp
-	clientProviderFunc clientProviderFunc
-	dbProviderFunc     dbProviderFunc
+	clientProviderFunc ClientProviderFunc
+	dbProviderFunc     DbProviderFunc
 	logger             *zap.Logger
-	client             dbClient
+	client             DbClient
 	db                 *sql.DB
 }
 
-var _ scraperhelper.Scraper = (*scraper)(nil)
+func NewScraper(id config.ComponentID, query Query, scrapeCfg scraperhelper.ScraperControllerSettings, logger *zap.Logger, providerFunc DbProviderFunc, clientProviderFunc ClientProviderFunc) *Scraper {
+	return &Scraper{
+		id:                 id,
+		query:              query,
+		scrapeCfg:          scrapeCfg,
+		logger:             logger,
+		dbProviderFunc:     providerFunc,
+		clientProviderFunc: clientProviderFunc,
+	}
+}
 
-func (s scraper) ID() config.ComponentID {
+var _ scraperhelper.Scraper = (*Scraper)(nil)
+
+func (s Scraper) ID() config.ComponentID {
```

Comment:
Can we make these all pointer receivers while we're in here? (this was my mistake -- there's just one at the moment and mixing pointer and non-pointer is bad, apparently)

Reply:
Done!